callback_plugins/custom_logger.py

```diff
@@ -1,92 +1,274 @@
+# (c) 2016 Matt Clay <matt@mystile.com>
+# (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```

Comment:
Copy/pasting the existing junitXML plugin is not the way to go.

Either try to contribute upstream to it, or inherit from the existing class here to make changes.

There MIGHT be an issue with using a GPL-licensed library to provide the base class, since this project is not GPL licensed, nor do we plan to adopt GPL v3 as the license.

Reply:
Yet another alternative would be to record the results and then render the XML at the end using a jinja2 template, so that there's no back-tracking through the document. However, this would use more memory.